dnSpy/dnSpy/Documents/Tabs/DefaultDocumentList.cs

```diff
@@ -48,6 +49,60 @@ sealed class DefaultDocumentList {
 		public void Add(DsDocumentInfo file) => files.Add(file);
 	}
 
+	sealed class DotnetReferenceFileFinder {
+		readonly CancellationToken cancellationToken;
+		readonly List<DefaultDocumentList> allLists;
+
+		public DotnetReferenceFileFinder(CancellationToken cancellationToken) {
+			this.cancellationToken = cancellationToken;
+			allLists = new List<DefaultDocumentList>();
+		}
+
+		public IEnumerable<DefaultDocumentList> AllFiles => allLists.ToArray();
+
+		public void Find() {
+			cancellationToken.ThrowIfCancellationRequested();
+
+			if (!TryGetRuntimeFolder(out var runtimesFolder)) {
+				return;
+			}
+
+			foreach (var appDir in Directory.EnumerateDirectories(runtimesFolder)) {
+				foreach (var versionDir in Directory.EnumerateDirectories(appDir)) {
+					var appName = $"{Path.GetFileName(appDir)} {Path.GetFileName(versionDir)}";
+					allLists.Add(new DefaultDocumentList(appName, Directory.GetFiles(versionDir, "*.dll").Where(FilterFiles).Select(DsDocumentInfo.CreateDocument)));
+				}
+			}
+		}
+
+		static bool FilterFiles(string file) {
+
+			var fileName = Path.GetFileName(file);
+
+			//Here, only DLLs that are obviously not .NET assemblies are filtered out; other files are not filtered, and should not be affected, so to speak.
+			if (fileName.StartsWith("api-ms-win")) {
+				return false;
+			}
+
+			return true;
+		}
+
+		static bool TryGetRuntimeFolder([NotNullWhen(true)] out string path) {
+			path = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles), @"dotnet\shared");
+			if (Directory.Exists(path)) {
+				return true;
+			}
+
+			//The previous dotnet installation was in the x86 folder
+			path = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ProgramFilesX86), @"dotnet\shared");
+			if (Directory.Exists(path)) {
+				return true;
+			}
+
+			return false;
+		}
```

Comment:
I think instead of using this method, it would be better to use `DotNetPathProvider` as it is already capable of locating and identifying the versions of the various .NET installations on the computer. Of course, the `DotNetPathProvider` class would need to be extended to output all the raw paths rather than the user having to request them by version/bitness.